### PR TITLE
r/network_acl + default_network_acl - add arn attribute + validations

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #13624, #13527

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_network_acl: add `arn` attribute
resource_aws_default_network_acl: add `arn` attribute

resource_aws_network_acl: add plan time validations for `from_port`, `to_port`, `rule_no`, `action`, `cidr_block`, `ipv6_cidr_block` for both `ingress` and `egress` blocks

resource_aws_default_network_acl: add plan time validations for `from_port`, `to_port`, `rule_no`, `action`, `cidr_block`, `ipv6_cidr_block` for both `ingress` and `egress` blocks
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSDefaultNetworkAcl_|TestAccAWSNetworkAcl_'
--- PASS: TestAccAWSDefaultNetworkAcl_basicIpv6Vpc (73.87s)
--- PASS: TestAccAWSDefaultNetworkAcl_deny_ingress (70.44s)
--- PASS: TestAccAWSDefaultNetworkAcl_withIpv6Ingress (71.89s)
--- PASS: TestAccAWSDefaultNetworkAcl_SubnetRemoval (140.85s)
--- PASS: TestAccAWSDefaultNetworkAcl_SubnetReassign (165.45s)
--- PASS: TestAccAWSNetworkAcl_basic (76.22s)
--- PASS: TestAccAWSNetworkAcl_disappears (72.56s)
--- PASS: TestAccAWSNetworkAcl_Egress_ConfigMode (164.35s)
--- PASS: TestAccAWSNetworkAcl_Ingress_ConfigMode (170.21s)
--- PASS: TestAccAWSNetworkAcl_EgressAndIngressRules (72.22s)
--- PASS: TestAccAWSNetworkAcl_OnlyIngressRules_basic (87.92s)
--- PASS: TestAccAWSNetworkAcl_OnlyIngressRules_update (134.21s)
--- PASS: TestAccAWSNetworkAcl_OnlyEgressRules (76.01s)
--- PASS: TestAccAWSNetworkAcl_SubnetChange (149.03s)
--- PASS: TestAccAWSNetworkAcl_Subnets (153.34s)
--- PASS: TestAccAWSNetworkAcl_SubnetsDelete (138.82s)
--- PASS: TestAccAWSNetworkAcl_ipv6Rules (88.37s)
--- PASS: TestAccAWSNetworkAcl_ipv6ICMPRules (67.22s)
--- PASS: TestAccAWSNetworkAcl_ipv6VpcRules (73.33s)
--- PASS: TestAccAWSNetworkAcl_espProtocol (72.58s)
--- PASS: TestAccAWSDefaultNetworkAcl_basic (74.51s)
--- PASS: TestAccAWSNetworkAcl_CaseSensitivityNoChanges (87.91s)
```

### CODE REVIEW

File: aws/resource_aws_default_network_acl.go [hunk 1]
```diff
@@ -64,32 +68,41 @@ func resourceAwsDefaultNetworkAcl() *schema.Resource {
...
 						"protocol": {
 							Type:     schema.TypeString,
 							Required: true,
 						},
 						"cidr_block": {
-							Type:     schema.TypeString,
-							Optional: true,
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.IsCIDR,
```
Comment (bflad) [inline aws/resource_aws_default_network_acl.go:100]: I think we need to allow empty strings in the cidr_block/ipv6_cidr_block attributes for Terraform 0.11 and earlier.

File: aws/resource_aws_default_network_acl.go [hunk 2]
```diff
@@ -64,32 +68,41 @@ func resourceAwsDefaultNetworkAcl() *schema.Resource {
...
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IsPortNumberOrZero,
 						},
 						"rule_no": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(1, 32766),
```
Comment (bflad) [inline aws/resource_aws_default_network_acl.go:83]: Nit: Doesn't need to be this PR, but we should probably create a constant for this, e.g.

```go
const (
	// Maximum number for EC2 Network ACL Rules. The range 32767 to 65535 is reserved for internal use.
	Ec2NetworkAclRuleNumberMaximum = 32766
)
```

We'll need to figure out the best place for these in the future when things are broken into separate packages.

File: website/docs/index.html.markdown [hunk 3]
```diff
@@ -276,6 +277,7 @@ for more information about connecting to alternate AWS endpoints or AWS compatib
   - [`aws_key_pair` resource](/docs/providers/aws/r/key_pair.html)
   - [`aws_launch_template` data source](/docs/providers/aws/d/launch_template.html)
   - [`aws_launch_template` resource](/docs/providers/aws/r/launch_template.html)
+  - [`aws_network_acl` resource](/docs/providers/aws/r/launch_template.html)
```
Comment (bflad) [inline website/docs/index.html.markdown:280]: Copypasta 🍝 

```suggestion
  - [`aws_network_acl` resource](/docs/providers/aws/r/network_acl.html)  
```

File: aws/resource_aws_default_network_acl.go (hunk 1 above)
Comment (DrFaust92) [reply aws/resource_aws_default_network_acl.go]: done

File: website/docs/r/default_network_acl.html.markdown [hunk 4]
```diff
@@ -182,6 +182,7 @@ can resume managing them via the AWS Console.
 In addition to all arguments above, the following attributes are exported:
 
 * `id` - The ID of the Default Network ACL
+* `id` - The ARN of the Default Network ACL
```
Comment (bflad) [inline website/docs/r/default_network_acl.html.markdown:185]:
```suggestion
* `arn` - The ARN of the Default Network ACL
```

**Pull request merged**